packages/twenty-server/src/engine/core-modules/auth/token/services/email-verification-token.service.ts

```diff
@@ -0,0 +1,103 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+
+import crypto from 'crypto';
+
+import { addMilliseconds } from 'date-fns';
+import ms from 'ms';
+import { Repository } from 'typeorm';
+
+import {
+  AppToken,
+  AppTokenType,
+} from 'src/engine/core-modules/app-token/app-token.entity';
+import { AuthToken } from 'src/engine/core-modules/auth/dto/token.entity';
+import {
+  EmailVerificationException,
+  EmailVerificationExceptionCode,
+} from 'src/engine/core-modules/email-verification/email-verification.exception';
+import { EnvironmentService } from 'src/engine/core-modules/environment/environment.service';
+
+@Injectable()
+export class EmailVerificationTokenService {
+  constructor(
+    @InjectRepository(AppToken, 'core')
+    private readonly appTokenRepository: Repository<AppToken>,
+    private readonly environmentService: EnvironmentService,
+  ) {}
+
+  async generateToken(userId: string, email: string): Promise<AuthToken> {
+    const expiresIn = this.environmentService.get(
+      'EMAIL_VERIFICATION_TOKEN_EXPIRES_IN',
+    );
+    const expiresAt = addMilliseconds(new Date().getTime(), ms(expiresIn));
+
+    const plainToken = crypto.randomBytes(32).toString('hex');
+    const hashedToken = crypto
+      .createHash('sha256')
+      .update(plainToken)
+      .digest('hex');
+
+    const verificationToken = this.appTokenRepository.create({
+      userId,
+      expiresAt,
+      type: AppTokenType.EmailVerificationToken,
+      value: hashedToken,
+      context: { email },
```

Comment:
FYI: In the future, we will remove the context column and save the email in the payload of the JWT token.

Reply:
Thanks for letting me know!